pkg/elle/core/core.go

```diff
@@ -54,22 +79,142 @@ func (c *CycleExplainer) renderCycleExplanation(explainer DataExplainer, circle
 	panic("impl me")
 }
 
-//  A function which takes a history and returns a {:graph, :explainer, :anomalies} map; e.g. realtime-graph.
-type Analyzer func(history History) (Anomalies, DirectedGraph, DataExplainer)
+// Analyzer is a function which takes a history and returns a {:graph, :explainer} map; e.g. realtime-graph.
+type Analyzer func(history History) (DirectedGraph, DataExplainer)
 
 // RealtimeGraph analyzes real-time
-func RealtimeGraph(history History) (Anomalies, DirectedGraph, DataExplainer) {
+func RealtimeGraph(history History) (DirectedGraph, DataExplainer) {
 	panic("implement me")
 }
 
+// ProcessExplainer ...
+type ProcessExplainer struct{}
+
+// ExplainPairData explain pair data
+func (e ProcessExplainer) ExplainPairData() interface{} {
+	panic("impl me")
+}
+
+// RenderExplanation render explanation
+func (e ProcessExplainer) RenderExplanation() string {
+	panic("impl me")
+}
+
+// ProcessOrder find dependencies of a process
+func ProcessOrder(history History, process int) DirectedGraph {
+	var (
+		processHistory History
+		graph          DirectedGraph
+	)
+
+	for _, op := range history {
+		if op.Process == process {
+			processHistory = append(processHistory, op)
+		}
+	}
+
+	for i := 0; i < len(processHistory)-1; i++ {
+		op1, op2 := processHistory[i], processHistory[i+1]
+		graph.Link(Vertex{op1}, []Vertex{{op2}}, Process)
+	}
+	return *graph.Fork()
+}
+
 // ProcessGraph analyzes process
-func ProcessGraph(history History) (Anomalies, DirectedGraph, DataExplainer) {
-	panic("implement me")
+func ProcessGraph(history History) (DirectedGraph, DataExplainer) {
+	var (
+		okHistory = history.FilterType(OpTypeOk)
+		processes map[int]struct{}
+		graphs    []DirectedGraph
+	)
+
+	for _, op := range okHistory {
+		if op.Process != 0 {
```

Comment:
Why filter out Process 0? what's Process 0 means?

Reply:
Process should not be 0 and 0 will be the default value when `Op` struct inits. Filter out Process with value 0 to exclude the operations which don't have a Process value.

Reply:
How about change Process type to *int?